### PR TITLE
feat: bulk archive usage warning

### PR DESCRIPTION
**Pull request by andreas-unleash**
Adds a warning when about to archive features that have lastSeenAt of less than a week (green usage)

Closes # [1-224](https://linear.app/unleash/issue/1-1224/bulk-edit-show-last-seen-usage-warning)
Screenshot 2023-08-08 at 15 10 26

### CODE REVIEW

File: frontend/src/component/project/Project/ProjectFeatureToggles/ProjectFeaturesBatchActions/ArchiveButton.tsx [hunk 1]
```diff
@@ -1,23 +1,52 @@
...
 interface IArchiveButtonProps {
     projectId: string;
-    features: string[];
+    featureIds: string[];
+    features: FeatureSchema[];
 }
 
+const DEFAULT_USAGE_THRESHOLD_DAYS = 7;
+
+const useFeatureUsage = () => {
```
Comment (kwasniew) [inline frontend/src/component/project/Project/ProjectFeatureToggles/ProjectFeaturesBatchActions/ArchiveButton.tsx:19]: why do we have useXYZ convention for read hooks. Isn't it a regular function? Also why do we have a function that creates another function? Can we drop the outermost function?

`const isFeatureInUser = (feature?: FeatureSchema) => ...`

File: frontend/src/component/project/Project/ProjectFeatureToggles/ProjectFeaturesBatchActions/ArchiveButton.tsx [hunk 2]
```diff
@@ -1,23 +1,52 @@
...
     projectId,
+    featureIds,
     features,
 }) => {
     const { refetch } = useProject(projectId);
     const [isDialogOpen, setIsDialogOpen] = useState(false);
     const { trackEvent } = usePlausibleTracker();
+    const featureIsInUse = useFeatureUsage();
+
+    const featuresWithUsage = useMemo(() => {
```
Comment (kwasniew) [inline frontend/src/component/project/Project/ProjectFeatureToggles/ProjectFeaturesBatchActions/ArchiveButton.tsx:40]: More idiomatic JS/TS:
```
    const featuresWithUsage = useMemo(() => {
        return featureIds.filter(name => {
            const feature = features.find(f => f.name === name);
            return isFeatureInUser(feature);
        });
    }, [JSON.stringify(features), featureIds]);
```

**Review by kwasniew — COMMENTED** [review]
Left some comments inline

File: frontend/src/component/project/Project/ProjectFeatureToggles/ProjectFeaturesBatchActions/ArchiveButton.tsx (hunk 1 above)
Comment (andreas-unleash) [reply frontend/src/component/project/Project/ProjectFeatureToggles/ProjectFeaturesBatchActions/ArchiveButton.tsx:19]: yup makes sense

**Review by kwasniew — APPROVED** [review]

**Pull request merged**